flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/SuccessfulCheckpoint.java

```diff
@@ -19,29 +19,28 @@
 package org.apache.flink.runtime.checkpoint;
 
 import org.apache.flink.api.common.JobID;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
+import java.io.Serializable;
+import java.util.ArrayList;
 import java.util.List;
 
 /**
  * A successful checkpoint describes a checkpoint after all required tasks acknowledged it (with their state)
  * and that is considered completed.
  */
-public class SuccessfulCheckpoint {
-	
-	private static final Logger LOG = LoggerFactory.getLogger(SuccessfulCheckpoint.class);
-	
+public class SuccessfulCheckpoint implements Serializable {
+
+	private static final long serialVersionUID = -8360248179615702014L;
+
 	private final JobID job;
 	
 	private final long checkpointID;
 	
 	private final long timestamp;
 	
-	private final List<StateForTask> states;
-
+	private final ArrayList<StateForTask> states;
```

Comment:
Because of Serializbility. SuccessfulCheckpoint has been changed to be Serializable. I figured this is OK, because it is an "internal" interface and the only usage creates and instance with ArrayList.